### DISCUSSION TITLE
Bisherige Ladedauer - API Key?

### DISCUSSION THREAD

**Discussion by Obo-Ron** [goecharger/go-eCharger-API-v2#150] [Q&A]
Hallo, in der App wird ja die Ladedauer aktuell immer mit angezeigt.
Gibt es dafür einen API Key, den ich bisher nicht gefunden habe oder muss ich das selbst zusammen bauen? Danke schon mal.

**Comment by 0xFEEDC0DE64** [comment]
Api key "cdi" charge duration info

kann null sein, wenn gerade nicht geladen wird,
oder ein json object sein.

das json object hat die keys "type" und "value":
wenn "type"==0, dann läuft die Ladung gerade, die Dauer wird relativ zur rbt reboot time als start der ladung ausgedrückt, `rbt-cdi.value` ist die ladedauer in millisekunden.
wenn "type"==1, dann ist die Ladung gerade pausiert, und die bisherige Ladedauer wird in "value" in millisekunden ausgedrückt (ohne rechnerei von rbt)

Dieses komplizierte Vorgehen spart bei uns Traffic ein, weil wir nicht jede Sekunde cdi übertragen müssen und anhand der rbt time eine Liveanzeige gebildet werden kann.
